sql-plugin/src/main/330+/scala/com/nvidia/spark/rapids/shims/Spark33XShims.scala

```diff
@@ -318,19 +318,116 @@ trait Spark33XShims extends Spark321PlusShims with Spark320PlusNonDBShims {
+        }),
+      GpuOverrides.expr[Add](
+        "Addition",
+        ExprChecks.binaryProjectAndAst(
+          TypeSig.implicitCastsAstTypes,
+          TypeSig.gpuNumeric + TypeSig.ansiInterval, TypeSig.numericAndInterval,
+          ("lhs", TypeSig.gpuNumeric + TypeSig.ansiInterval,
+              TypeSig.numericAndInterval),
+          ("rhs", TypeSig.gpuNumeric + TypeSig.ansiInterval,
+              TypeSig.numericAndInterval)),
+        (a, conf, p, r) => new BinaryAstExprMeta[Add](a, conf, p, r) {
+          private val ansiEnabled = SQLConf.get.ansiEnabled
+
+          override def tagSelfForAst(): Unit = {
+            if (ansiEnabled && GpuAnsi.needBasicOpOverflowCheck(a.dataType)) {
+              willNotWorkInAst("AST Addition does not support ANSI mode.")
+            }
+          }
+
+          override def convertToGpu(lhs: Expression, rhs: Expression): GpuExpression =
+            GpuAdd(lhs, rhs, failOnError = ansiEnabled)
+        }),
+      GpuOverrides.expr[Subtract](
+        "Subtraction",
+        ExprChecks.binaryProjectAndAst(
+          TypeSig.implicitCastsAstTypes,
+          TypeSig.gpuNumeric + TypeSig.ansiInterval, TypeSig.numericAndInterval,
+          ("lhs", TypeSig.gpuNumeric + TypeSig.ansiInterval,
+              TypeSig.numericAndInterval),
+          ("rhs", TypeSig.gpuNumeric + TypeSig.ansiInterval,
+              TypeSig.numericAndInterval)),
+        (a, conf, p, r) => new BinaryAstExprMeta[Subtract](a, conf, p, r) {
+          private val ansiEnabled = SQLConf.get.ansiEnabled
+
+          override def tagSelfForAst(): Unit = {
+            if (ansiEnabled && GpuAnsi.needBasicOpOverflowCheck(a.dataType)) {
+              willNotWorkInAst("AST Subtraction does not support ANSI mode.")
+            }
+          }
+
+          override def convertToGpu(lhs: Expression, rhs: Expression): GpuExpression =
+            GpuSubtract(lhs, rhs, ansiEnabled)
+        }),
+      GpuOverrides.expr[Alias](
+        "Gives a column a name",
+        ExprChecks.unaryProjectAndAstInputMatchesOutput(
+          TypeSig.astTypes,
+          (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.MAP + TypeSig.ARRAY + TypeSig.STRUCT
+              + TypeSig.DECIMAL_128 + TypeSig.ansiInterval).nested(),
+          TypeSig.all),
+        (a, conf, p, r) => new UnaryAstExprMeta[Alias](a, conf, p, r) {
+          override def convertToGpu(child: Expression): GpuExpression =
+            GpuAlias(child, a.name)(a.exprId, a.qualifier, a.explicitMetadata)
         })
     ).map(r => (r.getClassFor.asSubclass(classOf[Expression]), r)).toMap
     super.getExprs ++ map
   }
 
   // 330+ supports DAYTIME interval types
   override def getExecs: Map[Class[_ <: SparkPlan], ExecRule[_ <: SparkPlan]] = {
-    val _gpuCommonTypes = TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_64
+    val _gpuCommonTypes = TypeSig.commonCudfTypes + TypeSig.NULL
```

Comment:
Where is the Decimal Type Sig ?

Reply:
It's a refactor. The following references are all like: 
_gpuCommonTypes + TypeSig.DECIMAL_128, 
And TypeSig.DECIMAL_128 will overwrite  TypeSig.DECIMAL_64,
so can be safely deleted.